lib/kramdown/converter/html.rb

```diff
@@ -46,6 +46,14 @@ def initialize(root, options)
         @toc_code = nil
         @indent = 2
         @stack = []
+
+        @highlighter = options[:syntax_highlighter]
+        if @highlighter == :rouge
+          # assign a frozen string instead of converting Symbol to String for default use-case.
+          @highlighter = "rouge"
+        else
+          @highlighter = @highlighter.to_s
+        end
```

Comment:
Even if rouge *is* the default, I don't like hard-coding it this way.

Storing the stringified syntax highlighter is probably only useful when *many* code/pre sections exist. Otherwise I think the memory/cpu usage is negligible.

Reply:
*many* here is subjective..
I just ran this branch against the latest commit on `jekyll/jekyll`. Its docs site returned a total of `2894` calls to the `Converter::Html#add_syntax_highlighter_to_class_attr` method.
(*each call would convert `:rouge` to `"rouge"`*)

Is that *many* or just *not that many*..?

Reply:
Hmm... I think the way it is now is a good compromise and I hope you are okay with it. There are also [discussions](https://bugs.ruby-lang.org/issues/16153) going on whether `Symbol#to_s` should return a frozen string in the future.

Reply:
Yes, I'm okay with the *current* diff. :)